sdk/iot/core/README.md

```diff
@@ -28,7 +71,8 @@ void my_telemetry_func()
 
   //Allocate a span to put the telemetry topic
   uint8_t telemetry_topic_buffer[64];
-  az_span topic_span = AZ_SPAN_FROM_BUFFER(telemetry_topic_buffer);
+  az_span topic_span = az_span_init(telemetry_topic_buffer, 0,

```

Comment:
It looks like the same before/after this change. Why don't we use `AZ_SPAN_FROM_BUFFER` directly?

Reply:
On Linux it would be fine but it won't compile on Windows MSVC. [Example here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=291075&view=logs&j=e05b29e1-8ab3-5e14-836c-84b55b312f66&t=879b0a1c-774d-5a15-eb9c-e1b448929f69&l=83) at the gate where it fails.